builtin/credential/jwt/path_oidc.go

```diff
@@ -288,8 +296,13 @@ func (b *jwtAuthBackend) pathCallback(ctx context.Context, req *logical.Request,
 		return nil, errwrap.Wrapf("error getting provider for login operation: {{err}}", err)
 	}
 
-	var rawToken oidc.IDToken
+	oidcCtx, err := b.createCAContext(ctx, config.OIDCDiscoveryCAPEM)
```

Comment:
Out of curiosity, given we keep the same timeouts &c, why not reuse `ctx` here and simply add the value to the existing context we use elsewhere?

Is this an existing pattern that I'm missing?

Reply:
I guess the issue is what happens when we cancel a context? Would we want to cancel through to `caCtx` (or if it was cancelled separately, would we want to cancel the parent too)?

Reply:
I think it could be done that way but on the other hand it's probably a small amount of overhead to create and tear down a context, especially compared with everything else that's happening, so it's probably not worth it to cache the context.

Reply:
Yeah, fair.